waspc/src/Wasp/Generator/ServerGenerator.hs

```diff
@@ -252,3 +257,32 @@ patchesRequiredByPassport spec =
     ]
   where
     maybeAuth = AS.App.auth $ snd $ getApp spec
+
+-- Allows us to make specific changes to dependencies of our dependencies.
+-- This is helpful if something broke in later versions, etc.
+-- Ref: https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides
+getPackageJsonOverrides :: [Aeson.Value]
+getPackageJsonOverrides = map buildOverrideData (designateLastElement overrides)
+  where
+    overrides :: [(String, String, String)]
+    overrides =
+      [ -- sodium-native > 3.3.0 broke deploying on Heroku.
+        -- Ref: https://github.com/sodium-friends/sodium-native/issues/160
+        ("secure-password", "sodium-native", "3.3.0")
+      ]
+
+    -- NOTE: We must designate the last element so the JSON template can omit the final comma.
+    buildOverrideData :: (String, String, String, Bool) -> Aeson.Value
+    buildOverrideData (packageName, dependencyName, dependencyVersion, lastElement) =
+      object
+        [ "packageName" .= packageName,
+          "dependencyName" .= dependencyName,
+          "dependencyVersion" .= dependencyVersion,
+          "last" .= lastElement
+        ]
+
+    designateLastElement :: [(String, String, String)] -> [(String, String, String, Bool)]
```

Comment:
Has to be some cleaner way to do this, but couldn't quickly think of it.

Reply:
Yeah, true. I was thinking `zip` as well, but figured the shape got a bit stranger with that at the end. Plus, the fact we need the _last_ to be different would require doing a `True` first, then an infinite list of `False`, then reversing things a few times.

Unfortunately, if we leave a trailing comma, VS Code will put a red line under it since it is not valid JSON.

I will just go ahead and leave it as is for now since it is so localized. Thanks! 👍🏻